src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -140,6 +142,50 @@ static constexpr std::chrono::minutes MAX_LEDGER_AGE_ACQUIRE {1};
 // Don't acquire history if write load is too high
 static constexpr int MAX_WRITE_LOAD_ACQUIRE {8192};
 
+// Helper function for LedgerMaster::doAdvance()
+// Returns the minimum ledger sequence in SQL database, if any.
+boost::optional<LedgerIndex>
+minSqlSeq(Application& app)
+{
+    boost::optional<LedgerIndex> seq;
+    auto db = app.getLedgerDB().checkoutDb();
+    *db << "SELECT MIN(LedgerSeq) FROM Ledgers", soci::into(seq);
+    return seq;
+}
+
+// Helper function for LedgerMaster::doAdvance()
+// Return true if candidateLedger should be fetched from the network.
+bool
+shouldAcquire (
+    std::uint32_t const currentLedger,
+    std::uint32_t const ledgerHistory,
+    boost::optional<LedgerIndex> minSeq,
+    std::uint32_t const lastRotated,
+    std::uint32_t const candidateLedger,
+    beast::Journal j)
+{
+    // Fetch ledger if:
+    // 1) it might be the current ledger, or
+    // 2) is within our configured history range, or
+    // 3) is greater than or equal to both:
+    //     a. the minimum persisted ledger or
+    //        the maximum possible sequence value, if no persisted ledger, and
+    //     b. minimum ledger that will be persisted as of the next
+    //        online deletion interval, or 1 if online deletion is disabled.
+    bool ret (candidateLedger >= currentLedger ||
```

Comment:
Can we please use sensible syntax for initializing this variable?

```
bool ret = candidateLedger >= currentLedger ||
    (currentLedger - candidateLedger) <= ledgerHistory ||
        candidateLedger >= std::max(
            minSeq.value_or(std::numeric_limits<LedgerIndex>::max()),
            lastRotated + 1));
```

Also, please consider the following:

```
bool ret = [&]()
{
    // Fetch ledger if it may be the current ledger
    if (candidateLedger >= currentLedger)
        return true;

    // Or if it is within our configured history range:
    if (currentLedger - candidateLedger <= ledgerHistory)
        return true;

    // Or it's greater than or equal to both:
    // - the minimum persisted ledger or the maximum possible
    //   sequence value, if no persisted ledger, and
    // - minimum ledger that will be persisted as of the next online
    //   deletion interval, or 1 if online deletion is disabled.
    return 
        candidateLedger >= std::max(
            minSeq.value_or(std::numeric_limits<LedgerIndex>::max()),
            lastRotated + 1);
}();
```

Reply:
fixed